tests/test_views.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+
+from flask import render_template_string, g
+
+from udata.core.dataset.factories import DatasetFactory
+
+from udata_recommendations.views import NB_RECOMMENDATIONS
+
+
+def render_hook(dataset):
+    return render_template_string(
+        '{{ hook("dataset.display.after-description") }}',
+        dataset=dataset
+    )
+
+
+@pytest.fixture
+def datasets():
+    return DatasetFactory.create_batch(3)
+
+
+@pytest.mark.frontend
+@pytest.mark.usefixtures('clean_db')
+@pytest.mark.usefixtures('app')
+class TestViews:
+    def test_view_no_dataset(self):
+        assert '' == render_hook(dataset=None)
+
+    def test_view_dataset_no_extras(self):
+        assert '' == render_hook(dataset=DatasetFactory(extras={}))
+
+    def test_view_dataset_with_recommendations(self, datasets):
+        g.lang_code = 'fr'
```

Comment:
I was surprised I needed to set this. Otherwise, I had an error coming from the cache while using the `dataset/card.html` template. `lang_code` was undefined.

Please tell me if there is a proper way to set this

Reply:
Well this is strange 🧐 I would expect the frontend fixture to do this. Never had this problem though, I think.